### PR TITLE
JS Fundamentals Part 4: Clarify instructions for TDD exercises

### PR DESCRIPTION
**Pull request by jernestmyers**
- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
- [x] The title of this PR follows the `where the change occurs: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

- [x] I have previewed all lesson file(s) included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly.
- [x] If one exists, I have linked a related open issue to this PR in Step 2 below
- [ ] If changes were requested, I have made them and re-requested a review from the maintainer (top of the right sidebar)

**1. Description of the Changes**
 

- Adopted the Assignment markdown format since it visually organizes the lesson better
- Linked only to the repo's README to keep learners from jumping straight to the exercises without proper set up

**2. Related Issue**

I will also be removing unnecessary text in the `javascript-exercises` repo's README.

### CODE REVIEW

**Comment by jernestmyers** [comment]
@twalton83 Apologies if this is the wrong etiquette, but I'm directly tagging you since we spoke about this issue over the weekend.

File: foundations/javascript_basics/fundamentals-4.md [hunk 1]
```diff
@@ -5,49 +5,47 @@
...
 
-We will teach you the art of actually writing these tests later in the course.  The following exercises have the tests already written out for you. All you have to do is read the specs and write the code that makes them pass!  The very first exercise \(`01-helloWorld`\) is intentionally very simple and walks you through the process of running the tests and making them pass.  
+We will teach you the art of actually writing these tests later in the course. The following assignment has the tests already written out for you. All you have to do is set up the testing environment, read the specs, and write the code that makes them pass!
 
-### Good Luck!
+### Assignment
```
Comment (01zulfi) [inline foundations/javascript_basics/fundamentals-4.md:30]: I believe the term "Practice" goes well for this section since it has users do exercises which is essentially "practicing".
Can we change "Assignment" to "Practice" here?

File: foundations/javascript_basics/fundamentals-4.md [hunk 2]
```diff
@@ -5,49 +5,47 @@
...
 
-We will teach you the art of actually writing these tests later in the course.  The following exercises have the tests already written out for you. All you have to do is read the specs and write the code that makes them pass!  The very first exercise \(`01-helloWorld`\) is intentionally very simple and walks you through the process of running the tests and making them pass.  
+We will teach you the art of actually writing these tests later in the course. The following assignment has the tests already written out for you. All you have to do is set up the testing environment, read the specs, and write the code that makes them pass!
```
Comment (01zulfi) [inline foundations/javascript_basics/fundamentals-4.md:28]: Since the section will be called "Practice" (as per my comment), "the following assignment" text needs to be changed as well.

File: foundations/javascript_basics/fundamentals-4.md [hunk 3]
```diff
@@ -5,49 +5,47 @@
...
+1. Set up the files and Jest by carefully following the directions in the [repository's README](https://github.com/TheOdinProject/javascript-exercises#readme).
+2. Now that you have cloned the repository and run `npm install` to install Jest, complete these exercises in the following order:
+   - helloWorld (This exercise is intentionally very simple to ensure that you have set up everything properly!)
+   - repeatString
+   - reverseString
+   - removeFromArray
+   - sumAll
+   - leapYears
+   - tempConversion
+3. See how your solution compares to the exercise's given solution. Solutions for the exercises can be found on the 'solutions' branch of the repo.
```
Comment (01zulfi) [inline foundations/javascript_basics/fundamentals-4.md:44]: Really love this addition. Can we reword it to put a little more emphasis to have a "working solution" instead of just a "solution", or something along the lines of "a solution that passes all the tests"? 

I ask this because I believe there's value in being as explicit as possible here, so that users only look at the given solutions once they have a solution that works for them.

I have a suggestion here, but feel free to change the wording if it makes the message better.

```suggestion
3. Once you have a working solution, see how it compares to the exercise's given solution. Solutions for the exercises can be found on the 'solutions' branch of the repo.
```

**Review by 01zulfi — CHANGES_REQUESTED** [review]
Thanks for making these changes and for fixing inconsistencies across the lesson 🚀 

I have made some comments below. If you have any questions, let me know.

File: foundations/javascript_basics/fundamentals-4.md (hunk 3 above)
Comment (jernestmyers) [reply foundations/javascript_basics/fundamentals-4.md:44]: I thought the same thing but for whatever reason didn't include it! I've tried inserting an emphasis that the working solution should pass all the tests, but it kept feeling/sounding too verbose. I've opted instead to use your suggestion.

**Review by twalton83 — APPROVED** [review]
Thanks for the contribution.
